### PR TITLE
fix: handle workspace errors

### LINKED ISSUE #3094: Fix error messages in workspaceXService
### Problem
The frontend does not always display the error messages sent from the backend, and even when it does, it does not display the `detail` that is sometimes sent along with it. 

### Definition of done
- [ ]  There are no uses of `displayError` in this XService. To replace the error snackbar, render `ErrorSummary` in the relevant components with the `dismissible` prop. This way, the user can expand the error detail if desired without the snackbar disappearing too soon.
- [ ]  There are no uses of `getFormHelpers` in the components that consume errors from this XService. Switch to `getFormHelpersWithError` even if no error. (We will eventually rename it to replace the original `getFormHelpers`.)
- [ ]  The XService always `assign`s the API error, rather than a hard-coded error message. If a hard-coded error message is needed as a fallback, send it as a prop to `ErrorSummary`.
- [ ]  Errors are only unpacked/transformed in `ErrorSummary` and `getFormHelpersWithError`.
- [ ]  Storybook has a story for when the error(s) are showing.

### PR DESCRIPTION
**Pull request by AbhineetJain**
This PR handles the errors shown on the Workspace (singular) page. We handle the following types of errors:

- get workspace
- check permissions
- get resources
- get builds
- build error
- build cancellation

## Subtasks

- [x] display the errors using the `ErrorSummary` component for the listed errors
- [x] add stories

Fixes #3094 

### get resources
Screen Shot 2022-08-01 at 1 35 54 PM

### get builds
Screen Shot 2022-08-01 at 1 35 42 PM

### build cancellation
Screen Shot 2022-08-01 at 1 35 30 PM

### build error
Screen Shot 2022-08-01 at 1 18 15 PM

### get workspace
Screen Shot 2022-08-01 at 1 36 51 PM

### CODE REVIEW

**Review by presleyp — APPROVED** [review]
The code looks great! Do you know why the Chromatic check is being skipped?

**Review by kylecarbs — APPROVED** [review]

**Comment by AbhineetJain** [comment]
@presleyp I published the stories manually here: https://624de63c6aacee003aa84340-uboeickwbe.chromatic.com/?path=/story/components-workspace--error

**Comment by presleyp** [comment]
Storybook makes it look like we're displaying an error message when a build is successfully canceled - is it supposed to be an error saying it failed to cancel? Or a success message about cancellation?

File: site/src/xServices/workspace/workspaceXService.ts [hunk 1]
```diff
@@ -410,9 +407,6 @@ export const workspaceMachine = createMachine(
         assign({
           cancellationMessage: undefined,
         }),
-      displayCancellationError: (context) => {
-        displayError(context.cancellationMessage)
```
Comment (AbhineetJain) [inline site/src/xServices/workspace/workspaceXService.ts:414]: @presleyp Good point about the nature of the cancellation message. This was being displayed as an error before, so I just went with it. Do you think we should update this code to `displaySuccess` and remove it from the error list?

Comment (presleyp) [reply site/src/xServices/workspace/workspaceXService.ts:414]: Yeah if it's coming from an `onDone` then that makes sense

**Review by presleyp — APPROVED** [review]

**Pull request merged**